### ISSUE TITLE
IS05 interaction with IS04 test_15: Issue in counting undefined channels

### ISSUE THREAD

**Issue by kaihagseth** [AMWA-TV/nmos-testing#817]
I get the following error message:
SDP 'channel-order' for Sender <uid> does not match channels its Source <uid>

The sender is configured as a source with four channels: one mono group, one group of one undefined channel, and one gorup of two undefined channels. 

SDP: 
...
a=rtpmap:97 L24/44100/4
a=fmtp:97 channel-order=SMPTE2110.(M, U01, U02); TSMODE=SAMP; TSDELAY=0
...

Source:
...
"channels": [
{
"label": "M1",
"symbol": "M1"
},
{
"label": "U01",
"symbol": "U01"
},
{
"label": "U01",
"symbol": "U01"
},
{
"label": "U02",
"symbol": "U02"
}
],
...

So if I am not mistaken, the source should show the undefined channels as a list of undefined channels in the group, while the SDP will list the groups. This seems to be the way it's working for other groupings at least. Is there an issue with the test or have we misunderstood the spec, and in that case, what should the channels array look like?

**Comment by garethsb** [comment]
It's possible there's an issue with the test when there are two undefined groups in a row, I don't remember testing that before. Will check.

But also your SDP looks like it has white space after the commas in the list of channel groups, which I don't think is valid?

**Comment by kaihagseth** [comment]
> But also your SDP looks like it has white space after the commas in the list of channel groups, which I don't think is valid?

You are right, thanks for the heads up! We resolved that issue, on but the reported issue persists. Thank you for checking!

**Comment by garethsb** [comment]
Yes, the testing tool translates your Source `channels` to `channel-order=SMPTE2110.(M,U03)` and reports no match. Will look for guidance on handling multiple Undefined groups.

**Comment by peterbrightwell** [comment]
ARG call today: still relevant

**Comment by jonathan-r-thorpe** [comment]
Hi @kaihagseth. Did you happen to have a to fix for this? If so a Pull Request to the master branch would be most appreciated.

**Comment by kaihagseth** [comment]
I think the fix in https://github.com/AMWA-TV/nmos-testing/pull/901 should fix this issue as well. I had actually forgot i had reported this in 2023.
Edit: I ran some tests and confirm this is no longer an issue with the inputs provided in the original post.
